### PR TITLE
 p2p: use logdist(s1, sha3(n, s2)) for bucket choosing 

### PR DESCRIPTION
**Pull request by sorpaas**
Please merge after #523.

This implements countermeasure 3 in
https://www.cs.bu.edu/~goldbe/projects/eclipseEth.pdf by changing
bucket choosing from logdist(self.nodeID, n) to logdist(s1, sha3(n,
s2)) where s1, s2 are random private hash (32 bytearray).

### CODE REVIEW

File: p2p/discover/table.go [hunk 1]
```diff
@@ -51,6 +51,22 @@ const (
 	seedMaxAge          = 5 * 24 * time.Hour
 )
 
+var (
+	logdistS1 = new(common.Hash)
+	logdistS2 = new(common.Hash)
+)
+
+func init() {
+	s1a := make([]byte, 32)
```
Comment (whilei) [inline p2p/discover/table.go]: RNG alert -- do we want to use `rand.Seed` here as well?

Comment (whilei) [reply p2p/discover/table.go]: Ah, nevermind. `crypto/rand` not `math/rand`

Comment (sorpaas) [reply p2p/discover/table.go]: @whilei We need to make sure every time geth starts, logdistS1 and logdistS2 are of different value. So unless in tests, we can't use Seed.

Comment (whilei) [reply p2p/discover/table.go]: Wait a second... or? Maybe I'm not understanding either the purpose of `s1a` or how the `crypto/rand.Rand` works, in a mini-experiment I get the same results each run `�4�PI��` https://play.golang.org/p/Huafx9xcjeF

Comment (sorpaas) [reply p2p/discover/table.go]: That looks deterministic on purpose or something... Can you try it in your local computer?

Comment (whilei) [reply p2p/discover/table.go]: You're right. Weird. Thanks.

```go
~/sandbox  ⟠ go run rand.go
Hello, playground p�ȕs��ݳd�ؑ��
~/sandbox  ⟠ go run rand.go
Hello, playground !!���DI}k�Bւ6
~/sandbox  ⟠ go run rand.go
Hello, playground ��'PIש7f�ח��
~/sandbox  ⟠ go run rand.go
Hello, playground o	�����Vc)�!�
~/sandbox  ⟠ go run rand.go
Hello, playground }��&�����
~/sandbox  ⟠ go run rand.go
Hello, playground �����?�[�j���I^�
~/sandbox  ⟠ go run rand.go
Hello, playground �#wmC Y��6�D�
~/sandbox  ⟠ cat rand.go
package main

import (
	"crypto/rand"
	"fmt"
)

func main() {
	k := make([]byte, 16)
	rand.Read(k)
	fmt.Println("Hello, playground", string(k))
}
```

Comment (tzdybal) [reply p2p/discover/table.go]: Yeah, go playground is deterministic. Date is always the same.

**Pull request merged**